client/rpc/block_converter.go

```diff
@@ -0,0 +1,146 @@
+package rpc
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cmn "github.com/tendermint/tendermint/libs/common"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+	"github.com/tendermint/tendermint/version"
+)
+
+type ResultBlock struct {
```

Comment:
Fun fact, we don't need to reimplement/redefine everything! Instead, define the wrapper types and embed the original struct. Then, include any fields you wish to override. But because amino doesn't comply with the JSON stdlib, you'll have to implement a custom marshaler.

e.g.

```go
// Header defines a wrapper around Tendermint's Header type overriding various
// fields.
type Header struct {
	// embed original type
	tmtypes.Header

	// override fields from original type
	ProposerAddress sdk.ValAddress `json:"proposer_address"`
}

// MarshalJSON implements the json.Marshaler interface. We do this because Amino
// does not respect the JSON stdlib embedding semantics.
func (h Header) MarshalJSON() ([]byte, error) {
	type headerJSON Header
	_h := headerJSON(h)

	return json.Marshal(_h)
}
```

Reply:
Thanks for the tip, I've implemented as you suggested. I also used the same method for `CommitSig`. Could you please check if that's ok and/or could be used elsewhere?